### ISSUE TITLE
Atmosphere + Tomcat + Long-Polling: Possible race condition resulting in corrupt server response

### ISSUE THREAD

**Issue by kalbert312** [Atmosphere/atmosphere#1791]
We using the following frameworks & tools for our web application
- GWT 2.7.0
- Atmosphere Framework 2.2.3
- Atmosphere Client 2.2.5
- Atmosphere Guice 2.2.0
- Atmosphere GWT 2.2.2-GWT27
- Tomcat 8.0.15
- Guice 4.0-beta4-8d88344

We seem to be experiencing something eerily similar to http://mail-archives.apache.org/mod_mbox/tomcat-users/201403.mbox/%3C531848A4.4000609@aura.cz%3E
where normal responses from the server (not Atmosphere related) either have 0 as the response body or the response body is truncated. 

I do not have specific conditions to reproduce it but it seems to occur when Atmosphere restarts its long-polling request (could be a coincidence, though). My method of reproduction is bombarding our server with requests non-stop for 10 minutes or so until we get a corrupted response.

With the tools listed above, I was usually able to reproduce it in about 2-4 minutes.
Substituting Tomcat with Jetty 9.2.5, I was not able to reproduce it after 10 minutes.
Using Tomcat and removing atmosphere from both the server and client side, I was not able to reproduce it after 10 minutes.

I know this is somewhat vague and perhaps circumstantial, but could there be an issue here with atmosphere, tomcat, and long-polling? We must support IE8 and 9 so we need long-polling support in our application.

Thanks for any help.

**Comment by jfarcand** [comment]
@kalbert312 Any update on that issue?

**Comment by kalbert312** [comment]
According to our QA guys, they cannot reproduce the issue we were having in IE9. It leads me to believe that somehow having the duplicate interceptors was part of the problem. I did not change the list of default interceptors nor did I disable it but I listed all that I wanted in my @AtmosphereService annotation. I think this caused certain interceptors to be added more than once as evidence by the log I posted where HeartbeatInterceptor wrote 2 log entries.

Thank you for your help and work :)
